logstash-core/lib/logstash/state_resolver.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: utf-8
+module LogStash
+  # In the beginning I was using this code as a method in the Agent class directly
+  # But with the plugins system I think we should be able to swap what kind of action would be run.
+  #
+  # Lets take the example of dynamic source, where the pipeline config and settings are located and
+  # managed outside of the machine.
+  class StateResolver
+    def initialize(metric)
+      @metric = metric
+    end
+
+    def resolve(pipelines, pipeline_configs)
+      actions = []
+
+      pipeline_configs.each do |pipeline_config|
+        pipeline = pipelines[pipeline_config.pipeline_id]
+
+        if pipeline.nil?
+          actions << LogStash::PipelineAction::Create.new(pipeline_config, @metric)
+        else
+          # TODO(ph): The pipeline should keep a reference to the original PipelineConfig
+          # and we could use straight comparison.
+          if pipeline_config.config_hash != pipeline.config_hash
```

Comment:
@jsvd At first, I was thinking about implementing a comparator on the `PIpelineConfig` object and use that here by comparing the two `pipeline_config`, In that case we can check the string and the settings. WDYT?